From 0c4244251899f8e88ab4b7da19afe85a1462c3d4 Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Thu, 1 Aug 2019 13:44:53 +0800
Subject: [PATCH 24/52] net: stmmac: free pci irqs during driver unloading

We need to free PCI MSIs after the associated ISRs have been released.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index 4712788..f76bf65 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -611,6 +611,7 @@ static void stmmac_pci_remove(struct pci_dev *pdev)
 		break;
 	}
 
+	pci_free_irq_vectors(pdev);
 	pci_disable_device(pdev);
 }
 
-- 
2.7.4

